test/release/examples/benchmarks/lcals/LCALSMain.chpl

```diff
@@ -101,7 +101,7 @@ module main {
     run_loop[LoopKernelID.FIND_FIRST_MIN] = runC_findFirstMin;
 
     if output_dirname.length > 0 then
-      mkdir(output_dirname, parents=true);
+      try! mkdir(output_dirname, parents=true);
```

Comment:
Having just run release-over-release performance gathering, it occurs to me that a downside of this edit (which I liked on first blush due to its precision) is that it will prevent getting performance on this test for past releases of Chapel.  :(  

This wouldn't be much better, but is something I've been curious about for other reasons: Do we still have a global compiler flag that can be used to dial down strictness?  How does it play w.r.t. the pragma?

Reply:
We don't currently have a global compiler flag to dial down the strictness. Recall that before, we had only --strict-errors and --no-strict-errors, where --no-strict-errors was the global default. But I removed that flag with my earlier PR. We could certainly add a flag for --fatal-errors e.g.. Do you think we should?

Reply:
Also, if we want this test to work release-over-release, we could just have it call the out error= version of mkdir and include its own halt call. That should work the same before & after all the error handling changes.

Reply:
Thanks for the info (I'd lost track of the flag).  W.r.t. adding a flag: I don't feel certain about it, but while musing over the "fine-grain control" conversation yesterday and possible usage modes, it seemed likely that one might be "I'm going to use explicit modules, but I really don't want to be strict about my error-handling (yet?) and don't want to have to use a pragma on each module," which might suggest supporting such a compiler flag.

I'm not sure whether --fatal-errors is the right name for it or not.  Intuitively, it seems to me that the flag would be saying something like "explicit modules are relaxed w.r.t. error-handling requirements."  But this may be another conversation that suggests "Brad needs to refresh his understanding of the three modes."

Reply:
> Also, if we want this test to work release-over-release, we could just have it call the out error= version of mkdir and include its own halt call.

That might be nice, but is it a temporary solution?  (i.e., are we planning on deprecating those interfaces in a future release?)

Reply:
I'm uncertain about how temporary it would be. I think we intend to deprecate those interfaces at some point, yes, but using an `error=` argument seems like a reasonable "for now" change here.

For the compiler flag, what do you think about `--halt-on-errors` ?

Reply:
OK, let's do the "for now" change since this is an important performance test.  In non-performance cases, though, I like leaning on the new features more (so wish we could here as well... but not enough I guess).

In English, is the behavior of this flag "halt on errors" or "halt on uncaught errors" or "..."?

Reply:
"Instead of having a compiler error on unhandled errors, halt at runtime if one comes up"

Reply:
--halt-on-unhandled-errors?
--unhandled-errors-halt?

The main hesitation I have with all three of the above suggestions (including yours) is that it almost sounds as though we're requesting a different runtime behavior for a given program when in fact we're really requesting a different compile-time behavior, aren't we?  Something like "permit errors to be unhandled and halt on them"?

(You can tell me to go back and read about the three modes again at any time that I'm talking nonsense).

Reply:
PR #7283 is about the lcals changes. I'll ask David to look at it.

Reply:
It's not exactly only compile-time or run-time, because it's a choice between compile-time failure for the code that doesn't handle errors vs. run-time error checking. I think it *is* different runtime behavior - checking for errors and halting if any are found. That is different run-time behavior than you'd get with marking a function 'throws' say (which is probably the most common way to consume the errors).

Reply:
Other potential flag names: `--lazy-errors` or `--prototype-errors`

Reply:
Of the ones we've discussed, ``--lazy-errors`` is most appealing to me in its conciseness and the broad net it casts.  Though it's not as concise, I also find ``--enable-lazy-errors`` or ``--permit-lazy-errors`` to be slightly clearer.  Might be time to get some opinions other than mine on this question.

Apart from name, is the feature easy to implement?  (I'm guessing so).  Is it correct to characterize that it would not change the behavior of anything that is making an effort to handle errors properly and successfully does so?

Reply:
Yes - easy to implement and also wouldn't change the behavior of anything making an effort to handle errors properly. And I think I'd implement it to just change the default for modules from "error mode relaxed" to "error mode fatal".

Reply:
That sounds right to me.  You're likely already on this, but I'd implement it, and then send out a mail or issue explaining it and containing a short poll with favorite names to see if there's a clear winner.